client/state/imports/site-importer/actions.js

```diff
@@ -0,0 +1,265 @@
+			startSiteImporterImport()
+		)
+	);
+
+	wpcom.wpcom.req
+		.post( {
+			path: `/sites/${ siteId }/site-importer/import-site?${ stringify( params ) }`,
+			apiNamespace: 'wpcom/v2',
+			formData: [
+				[ 'import_status', JSON.stringify( toApi( importerStatus ) ) ],
+				[ 'site_url', targetSiteUrl ],
+			],
+		} )
+		.then( response => {
+			// At this point we're assuming that an import is going to happen
+			// so we set the user's editor to Gutenberg in order to make sure
+			// that the posts aren't mangled by the classic editor.
+			if ( 'engine6' === engine ) {
+				dispatch( setSelectedEditor( siteId, 'gutenberg' ) );
+			}
+
+			dispatch(
+				withAnalytics(
+					recordTracksEvent( 'calypso_site_importer_start_import_success', trackingParams ),
+					siteImporterImportSuccessful( response )
+				)
+			);
+
+			defer( () => {
+				const data = fromApi( response );
+				// Note: We're creating the finishUploadAction using the locally generated ID here
+				// as opposed to the new import ID recieved in the API response.
+				const action = createFinishUploadAction( importerStatus.importerId, data );
+
+				Dispatcher.handleViewAction( action );
+
+				startMappingSiteImporterAuthors( {
+					importerStatus: data,
+					site,
+					targetSiteUrl,
+				} )( dispatch );
+			} );
+		} )
+		.catch( error => {
+			dispatch(
+				withAnalytics(
+					recordTracksEvent( 'calypso_site_importer_start_import_fail', trackingParams ),
+					siteImporterImportFailed( error )
+				)
+			);
+		} );
+};
+
+export const validateSiteIsImportable = ( { params, site, targetSiteUrl } ) => dispatch => {
+	const siteId = site.ID;
+
+	prefetchmShotsPreview( targetSiteUrl );
+
+	dispatch(
+		withAnalytics(
+			recordTracksEvent( 'calypso_site_importer_validate_site_start', {
+				blog_id: siteId,
+				site_url: targetSiteUrl,
+			} ),
+			startSiteImporterIsSiteImportable()
+		)
+	);
+
+	return wpcom.wpcom.req
+		.get( {
+			path: `/sites/${ siteId }/site-importer/is-site-importable?${ stringify( params ) }`,
+			apiNamespace: 'wpcom/v2',
+		} )
+		.then( response =>
+			dispatch(
+				withAnalytics(
+					recordTracksEvent( 'calypso_site_importer_validate_site_success', {
+						blog_id: siteId,
+						site_url: response.site_url,
+						supported_content: response.supported_content
+							.slice( 0 )
+							.sort()
+							.join( ', ' ),
+						unsupported_content: response.unsupported_content
+							.slice( 0 )
+							.sort()
+							.join( ', ' ),
+						site_engine: response.engine,
+					} ),
+					startSiteImporterIsSiteImportableSuccessful( response )
+				)
+			)
+		)
+		.catch( error =>
+			dispatch(
+				withAnalytics(
+					recordTracksEvent( 'calypso_site_importer_validate_site_fail', {
+						blog_id: siteId,
+						site_url: targetSiteUrl,
+					} ),
+					startSiteImporterIsSiteImportableFailed( error )
+				)
+			)
+		);
+};
+
+export const resetSiteImporterImport = ( { importStage, site, targetSiteUrl } ) => dispatch =>
+	dispatch(
```

Comment:
The `dispatch => dispatch` bit is redundant -- you can just return the `withAnalytics` return value. Or dispatch both actions separately -- then the thunk becomes useful.